src/gui/plugins/scene3d/Scene3D.cc

```diff
@@ -777,6 +777,23 @@ void IgnRenderer::HandleKeyPress(QKeyEvent *_e)
     this->dataPtr->mousePressPos = this->dataPtr->mouseEvent.Pos();
   }
 
+  // fullscreen
+  if (_e->key() == Qt::Key_F11)
+  {
+    if (ignition::gui::App()->findChild
+        <ignition::gui::MainWindow *>()->QuickWindow()->visibility() 
+        == QWindow::FullScreen)
+    {
+      ignition::gui::App()->findChild
+        <ignition::gui::MainWindow *>()->QuickWindow()->showMaximized();
```

Comment:
How about using `showNormal` here instead? For me, that's bringing the window to the state it had before going fullscreen.

```suggestion
        <ignition::gui::MainWindow *>()->QuickWindow()->showNormal();
```